subprojects/docs/src/docs/userguide/customTasks.xml

```diff
@@ -305,4 +305,181 @@
             </sample>
         </section>
     </section>
+    <section id="worker_api">
+        <title>The Worker API</title>
+        <note>
+            <para>
+                The Worker API is an <link linkend="feature_lifecycle">incubating</link> feature.
+            </para>
+        </note>
+        <para>
+            As can be seen from the discussion of <link linkend="incremental_tasks">incremental tasks</link>, the work that a task performs can be viewed as discrete units (i.e. a subset of inputs that are transformed to a certain subset of outputs).  Many times, these units of work are highly independent of each other, meaning they can be performed in any order and simply aggregated together to form the overall action of the task.  In a single threaded execution, these units of work would execute in sequence, however if we have multiple processors, it would be desirable to perform independent units of work concurrently.  By doing so, we can maximize the resources available at build time and complete the activity of the task faster.
+        </para>
+        <para>
+            The Worker API provides a mechanism for doing exactly this.  It allows for safe, concurrent execution of multiple items of work during a task action.  But the benefits of the Worker API are not confined to parallelizing the work of a task.  You can also configure a desired level of isolation such that work can be executed in an isolated classloader or even in an isolated process.  Furthermore, the benefits extend beyond even the execution of a single task.  Using the Worker API, Gradle can begin to execute tasks in parallel by default.  In other words, once a task has submitted its work to be executed asynchronously, and has exited the task action, Gradle can then begin the execution of other independent tasks in parallel, even if those tasks are in the same project.
```

Comment:
I'm going to defer this to a follow-up story to do some further improvements to these docs.